### ISSUE TITLE
Setting rule options inline in source code with flags

### ISSUE THREAD

**Issue by borekb** [palantir/tslint#4264]
[Rule flags](https://palantir.github.io/tslint/usage/rule-flags/) allow enabling and disabling rules but not setting / overriding values.

For example, we use single quotes for most of our TSX but in Jest tests, we're forced to use double quotes so at the top of our test files, I'd like to be able to do something like this (not sure about the syntax):

```tsx
// tslint:rule:quotemark:[true, "single", "jsx-double"]
```

Probably related:

- "Feature request: override rules per file / location" #126 – was really about enabling / disabling rules which is handled well by rule flags.
- "Proposal: Conditional lint configuration" #3447 – this might solve it on another level. I still think that an inline comment-based syntax could be useful though.

**Comment by JoshuaKGoldberg** [comment]
+1 to this idea in general. Very useful! However, rules are being moved to a more standardized, less-brittle options format:

```
    "semicolon": {
        options: ["always"],
    },
    "space-before-function-paren": {
        options: {
            anonymous: "never",
            asyncArrow: "always",
            constructor: "never",
            method: "never",
            named: "never",
        },
    },
    "trailing-comma": {
        options: {
            multiline: "always",
            singleline: "never",
        },
    }
```

How would this handle setting these fancy options?

**Comment by borekb** [comment]
Oh, I didn't know about the options format. I think that I'd just use the value of `options` like this:

```
// tslint:rule:semicolon:["always"]
// tslint:rule:space-before-function-paren:{anonymous: "never", asyncArrow: "always", constructor: "never", method: "never", named: "never"}
// tslint:rule:trailing-comma:{multiline: "always", singleline: "never"}
```

The middle one is long and ugly but for simpler rules, this could work quite well.

**Comment by borekb** [comment]
For reference, I tried to look whether ESLint has similar feature to possibly use their syntax / approach but it doesn't seem so. I didn't find an issue requesting it either. Is my use case too weird? :)

**Comment by reduckted** [comment]
>  Is my use case too weird? :)

Are you able to add a tslint.json file to the test directory that inherits the root config, but changes the settings for that particular rule? Or is it only some test files that you need to override the settings in?

**Comment by pburkindine** [comment]
Specifically, for max-file-line-count, I never ever want to disable this rule for a file (goodbye!). When making an exception, I want to increase the cap to a new value on a per-file basis

Something like

// tslint:configure:max-file-line-count [true, 400]

// tslint:reset:some-rule

**Comment by lonix1** [comment]
> Is my use case too weird? :)

NO

I have this problem in unit test files. There are rules I want to keep, but modify their behaviour. So specifying arguments is useful.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
